src/MyConnection.h

```diff
@@ -30,15 +35,21 @@ class MyConnection : boost::noncopyable {
     mysql_options(pConn_, MYSQL_OPT_LOCAL_INFILE, 0);
     // Default to UTF-8
     mysql_options(pConn_, MYSQL_SET_CHARSET_NAME, "UTF8");
-    if (groups != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP, groups.c_str());
-    if (default_file != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_FILE, default_file.c_str());
-
-
-    if (!mysql_real_connect(pConn_, host.c_str(), user.c_str(),
-        password.c_str(), db == "" ? NULL : db.c_str(), port,
-        unix_socket == "" ? NULL : unix_socket.c_str(), client_flag)) {
+    if ((!groups.isNull()) && (Rcpp::as<std::string>(groups) != ""))
```

Comment:
With regard to the casting of a `Rcpp::Nullable < str::string >`: I just tried `groups.get()`, but it doesn't work. I had already found that implicit casting (`groups.c_str()`) fails to build as well:

`error: ‘class Rcpp::Nullable<std::basic_string<char> >’ has no member named ‘c_str’`

Since `Rcpp::Nullable` is so new, there are not so many usage examples around yet either. I got this way of casting from some example code in issue RcppCore/Rcpp#394.

With regard to the checks on `groups` and `default_file` to prevent passing `""` to `mysql_options()`, I now realize that those checks were only there because the default values were `""`. Now that the default values are `NULL`, the check is indeed unnecessary. As far as I can see the current stable version passes on `""` without modification. So, I will remove those checks.